directives/date-range-picker/date-range-picker.directive.js

```diff
@@ -26,25 +28,39 @@
         function postlink(scope, elem, attrs, ctrls) {
             var ngModelCtrl = ctrls[0],
                 dateRangeCtrl = ctrls[1];
-            scope.notEqual = attrs.mustBeEqualMsg;
-            scope.noStart = attrs.startAndEndMsg;
 
             scope.dummyMinDate = new Date('1970-01-01');
             scope.displayPopup = function() {
                 return scope.templateType === 'popup';
             };
+            scope.displayError = displayError;
 
             popupSetup(scope);
 
-            scope.dateFormat = 'shortDate';
-            attrs.$observe('dateFormat', function(v) {
-                scope.dateFormat = v;
-            });
+            scope.dateFormat = $locale.DATETIME_FORMATS.shortDate;
 
             scope.setRangeClass = setRangeClass;
 
-            ngModelCtrl.$validators.empty = validateByValidDates;
-            ngModelCtrl.$validators.order = validateByValidOrder;
+            scope.defaultValidators = [
+              {
+                  key: 'order',
+                  message: attrs.invalidOrderMessage || 'StartDateMustBeEqualToOrEarlierThanEndDate',
+                  validate: _validateStartAndEndByOrder
+              },
+              {
+                  key: 'empty',
```

Comment:
it looks like this error can't be triggered anymore. The "lessThan7Days" rules is triggered first.